### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by williamscch**
In this PR I used Redux store to display books

- The books are being displayed from the Redux store using the `useSelector` method to consume the Redux state.

- In the React component responsible for adding new books, added functionality for the "Add new" button click which uses the` useDispacth` redux method to update the state as well as in the Delete button using the action creators.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
 import Navigation from './components/Navigation';
 import Categories from './components/Categories';
 import BooksPage from './components/BooksPage';
-// import { store } from './redux/books/books';
+import store from './redux/configureStore';
 
 const App = () => (
-  // <Provider store={store}>
-  <>
+  <Provider store={store}>
```
Comment (Mihndim2020) [inline src/App.js]: - [OPTIONAL] As it is a [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components.md) for this project and best practice, " `Import Redux Provider and your store in the main component of your React app.` ", The main React component (`App.js`) should be wrapped in Provider and assign store. Something like so: 
```
    import store from './redux/configureStore';

    ReactDOM.render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root')
    )
```

File: src/components/BooksList.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+// import PropTypes from 'prop-types';
```
Comment (Mihndim2020) [inline src/components/BooksList.js]: - [OPTIONAL] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
### Hi @williamscch, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (williamscch) [reply src/App.js]: Okay, I understand

File: src/components/BooksList.js (hunk 2 above)
Comment (williamscch) [reply src/components/BooksList.js]: Got it, Thank you

**Pull request merged**